Modules/_io/bytesio.c

```diff
@@ -404,15 +405,20 @@ read_bytes(bytesio *self, Py_ssize_t size)
     if (size > 1 &&
```

Comment:
I'm not sure about the assertion above.

Should it be ``assert((self->pos + size) <= self->string_size);`` ?

Reply:
I changed it and then `test_overseek()` failed. The test seeks to a position after the end of the buffer (which is apparently legal) and then calls `read()`, which ends up calling `read_bytes` with `size=0`, causing the assertion to fail.

Reply:
> The test seeks to a position after the end of the buffer (which is apparently legal)

Is it useful to create "holes" and write after the end?

Reply:
Yes, I think this corresponds to creating sparse files.